dids/didjwk.go

```diff
@@ -0,0 +1,117 @@
+package dids
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+
+	"github.com/tbd54566975/web5-go/crypto"
+	"github.com/tbd54566975/web5-go/crypto/dsa"
+	"github.com/tbd54566975/web5-go/jwk"
+)
+
+type newDIDJWKOptions struct {
+	keyManager  crypto.KeyManager
+	algorithmID string
+}
+
+type newDIDJWKOption func(o *newDIDJWKOptions)
```

Comment:
Verrrry interesting pattern... basically, the params are functions which take pointers to an instance of the options struct, so that calling `NewDIDJWK` supports "Options can be passed in any order"

But a byproduct is that the function has to start out with this bit of code

```golang
	o := newDIDJWKOptions{
		keyManager:  crypto.NewLocalKeyManager(),
		algorithmID: dsa.AlgorithmID.ED25519,
	}

	for _, opt := range opts {
		opt(&o)
	}
```

A bit counterintuitive to learn, but makes the DX from the outside in more dynamic... which, I'm not sure I love but perhaps that's a bias from my experience